src/Analyzer/FunctionNode.cpp

```diff
@@ -209,15 +209,17 @@ ASTPtr FunctionNode::toASTImpl(const ConvertToASTOptions & options) const
         function_ast->kind = ASTFunction::Kind::WINDOW_FUNCTION;
     }
 
+    const auto & arguments = getArguments();
     auto new_options = options;
+    const auto & argument_nodes = arguments.getNodes();
     /// To avoid surrounding constants with several internal casts.
-    if (function_name == "_CAST" && (*getArguments().begin())->getNodeType() == QueryTreeNodeType::CONSTANT)
+    if (function_name == "_CAST" && argument_nodes.size() > 0 && argument_nodes[0]->getNodeType() == QueryTreeNodeType::CONSTANT)
```

Comment:
Why can `_CAST` have no arguments?

Reply:
I assume it can be in case of malformed query. Normally we should throw an exception earlier, but in some cases we try to print such ASTs. The same with IN below that requires at least 2 arguments.

My current understanding is that we put into scope unresolved node and trying to print it in error message when we yet don't know that AST is invalid. Perhaps this should be fixed, but requires more investigation.

Ant actually I can imagine some hypothetical piece of code like that also should be fine:

```cpp
if (node->getArguments().size() != 2)
    throw Expection("Wrong number of args in {}", node->formatASTForErrorMessage());
```

Reply:
If the user writes it explicitly as `_CAST()` which is incorrect.